src/algorithms/peps_opt.jl

```diff
@@ -41,13 +86,36 @@ the CTMRG runs can be started on the converged environments of the previous opti
 step by setting `reuse_env` to true. Otherwise a random environment is used at each
 step. The CTMRG gradient itself is computed using the `gradient_alg` algorithm.
 """
-@kwdef struct PEPSOptimize{G}
-    boundary_alg::CTMRG = CTMRG()  # Algorithm to find boundary environment
-    optimizer::OptimKit.OptimizationAlgorithm = LBFGS(
-        4; maxiter=100, gradtol=1e-4, verbosity=2
-    )
-    reuse_env::Bool = true  # Reuse environment of previous optimization as initial guess for next
-    gradient_alg::G = GeomSum() # Algorithm to solve gradient linear problem
+struct PEPSOptimize{G}
+    boundary_alg::CTMRG
+    optimizer::OptimKit.OptimizationAlgorithm
+    reuse_env::Bool
+    gradient_alg::G
+
+    function PEPSOptimize(  # Inner constructor to prohibit illegal setting combinations
+        boundary_alg::CTMRG{S},
+        optimizer,
+        reuse_env,
+        gradient_alg::G,
+    ) where {S,G}
+        if gradient_alg isa GradMode
+            if S == :sequential && G.parameters[1] == :fixed
```

Comment:
I don't really like accessing the internals of a type like this. In particular, note that this would fail if you make a new subtype: `struct MyGradMode{T,F} <: GradMode{F} end`.
The more idiomatic way is something like:
```julia
iterscheme(alg::GradMode{F}) where {F} = F
```
which would then automatically extract the correct parameter for all subtypes

Reply:
The `.parameters` thing is a bad habit of mine! I should definitely write functions for that.